rust/lance-arrow/Cargo.toml

```diff
@@ -10,6 +10,9 @@ description = "Arrow Extension for Lance"
 keywords.workspace = true
 categories.workspace = true
 
+[lib]
+crate-type = ["cdylib", "rlib"]
```

Comment:
Why does this need to be `cdylib`?

Reply:
This is required for `wasm-pack`

Reply:
Okay. It's just hard to tell from this PR. In the future, consider adding more comments as well as a CI job to continue to verify WASM actually works with the crate.

Reply:
Sg. I will setup a wasm CI this weekend if i can find some bandwidth.